### PR TITLE
[4] Access patent data

### LINKED ISSUE #4: Access patent data
At least couple of options:
- Access Google Patent dataset (similar to previous AI x Genomics or ARIA work)
- Try out LogicMill platform

### PR DESCRIPTION
**Pull request by beingkk**
Closes #4 

# Description

Hey @RFOxbury, please see this small PR, adding module for fetching data from Google's BigQuery database

I've also written a few tests for the module (mostly for demonstration purposes). You can do the tests by running
```
pytest discovery_child_development/tests/test_utils_bigquery.py
```
Tests is something that is very good to do, but we seldom have the time. When I'm writing utils, I find it helps me a lot in figuring out how exactly the utils should behave. When you do have time, suggest looking into testing and test-driven development (and perhaps you've come across it already?)

# Instructions for Reviewer

In order to test the code in this PR you can try to run `notebooks/patents/test.py` (first, convert to .ipynb using jupytext)

# Checklist:

- [x] I have refactored my code out from `notebooks/`
- [x] I have checked the code runs
- [x] I have tested the code
- [x] I have run `pre-commit` and addressed any issues not automatically fixed
- [ ] I have merged any new changes from `dev`
- [x] I have documented the code
  - [x] Major functions have docstrings
  - [] Appropriate information has been added to `README`s
- [x] I have explained this PR above
- [x] I have requested a code review

### CODE REVIEW

**Comment by beingkk** [comment]
fyi @RFOxbury I branched off your branch for this one. Once you've completed all the adjustments and merged into `dev`, I'll be simply merging `dev` into this branch to get the updates.

File: requirements.txt [hunk 1]
```diff
@@ -11,4 +11,6 @@ scikit-learn==1.3.1
 sentence-transformers==2.2.2
 spacy==3.6.1
 en_core_web_sm @ https://github.com/explosion/spacy-models/releases/download/en_core_web_sm-3.6.0/en_core_web_sm-3.6.0-py3-none-any.whl
-umap-learn==0.5.4
\ No newline at end of file
+umap-learn==0.5.4
+google-cloud-bigquery==3.11
+git+https://github.com/nestauk/nesta_ds_utils.git#egg=nesta_ds_utils"[s3,viz]"
```
Comment (RFOxbury) [inline requirements.txt]: I think this should be
```
nesta_ds_utils[s3,viz] @ git+https://github.com/nestauk/nesta_ds_utils.git
```

File: discovery_child_development/tests/test_utils_bigquery.py
Comment (RFOxbury) [inline discovery_child_development/tests/test_utils_bigquery.py:1]: Some of these tests fail if you run
```
pytest discovery_child_development/tests/test_utils_bigquery.py
```
when the credentials file does not yet exist in `.credentials/` (ie you have not yet run `create_client()`) - not sure if that is a problem or not?

File: discovery_child_development/notebooks/patents/test.py
Comment (RFOxbury) [inline discovery_child_development/notebooks/patents/test.py:1]: This runs fine for me :)

**Review by RFOxbury — APPROVED** [review]

File: discovery_child_development/tests/test_utils_bigquery.py
Comment (beingkk) [reply discovery_child_development/tests/test_utils_bigquery.py:1]: Good catch, thanks @RFOxbury! I realise that those two tests didn't really make sense - so have removed one of them, and adjusted the other. That's what happens when you use ChatGPT to help you with writing unit tests :]

**Comment by beingkk** [comment]
Thanks a lot for the review!

**Pull request merged**